code/datums/components/_component.dm

```diff
@@ -294,10 +301,13 @@
 		return
 	var/comps = dc[/datum/component]
 	if(islist(comps))
-		for(var/I in comps)
-			target.TakeComponent(I)
+		for(var/datum/component/I in comps)
```

Comment:
there can be non-components in comps?

Reply:
Not sure, I just updated our components to the most recent version of /tg/. I'm not quite sure if they intentionally made it like that.